### PR TITLE
OSSMDOC-335: Document lack of support for canary upgrades.

### PR DESCRIPTION
**Pull request by JStickler**
Version(s): 4.6 - 4.11

Issue: [OSSMDOC-335](https://issues.redhat.com/browse/OSSMDOC-335) Document OSSM upgrades incl In Place vs Upstream Canary Deployments.

The documentation already includes lack of support for canary upgrades here:
https://docs.openshift.com/container-platform/4.11/service_mesh/v2x/ossm-vs-community.html#ossm-installation-upgrade_ossm-vs-istio

This PR adds the following:
A note about gateways being manged by OLM.
Link to docs preview:  http://file.bos.redhat.com/jstickle/OSSMDOC-335/service_mesh/v2x/ossm-vs-community.html#ossm-gateways_ossm-vs-istio 
Information about lack of support for canary upgrades to the Upgrade assembly.
Link to docs preview:  http://file.bos.redhat.com/jstickle/OSSMDOC-335/service_mesh/v2x/upgrading-ossm.html#upgrading-control-plane

Additional information:
Eng review: rcernich
QE review: mattmahoneyrh
Peer review: tmalove

### CODE REVIEW

File: modules/ossm-vs-istio.adoc [hunk 1]
```diff
@@ -126,7 +126,12 @@ You can deploy virtual machines to OpenShift using OpenShift Virtualization. The
 [id="ossm-gateways_{context}"]
 == Gateways
 
-{SMProductName} installs ingress and egress gateways by default. You can disable this in the SMCP using `spec.gateways.ingress.enabled=false` or `spec.gateways.egress.enabled=false`.
+{SMProductName} installs ingress and egress gateways by default. You can disable this in the `ServiceMeshControlPlane` (SMCP) resource by setting `spec.gateways.ingress.enabled=false` or `spec.gateways.egress.enabled=false`.
```
Comment (rcernich) [inline modules/ossm-vs-istio.adoc:129]: Not sure if it's worth mentioning that they can be disabled together by `spec.gateways.enabled=false`.

**Review by rcernich — APPROVED** [review]

File: modules/ossm-vs-istio.adoc (hunk 1 above)
Comment (JStickler) [reply modules/ossm-vs-istio.adoc:129]: One setting instead of two?  Yep, worth adding that.

**Comment by tsze-redhat** [comment]
Will it be better to change

"You cannot update from {SMProductName} 2.0 to 2.2 directly."
to
"You _should not_ update from {SMProductName} 2.0 to 2.2 directly"?

Thanks.

**Comment by mattmahoneyrh** [comment]
LGTM - tested and works as expected.

**Comment by JStickler** [comment]
@tsze-redhat  your suggestion goes against our style guide, IBM Style.
**should** Always prefer a more direct verb or way of saying something.  Omit "should" use "must" or "might' as appropriate.

Also we discussed this at some point, that upstream Istio only supports upgrading two versions at a time. Because each OSSM version contains multiple Istio versions, it's safer to go from one OSSM release to the next (which is equal to multiple Istio version upgrades).

File: modules/ossm-vs-istio.adoc [hunk 2]
```diff
@@ -126,7 +126,16 @@ You can deploy virtual machines to OpenShift using OpenShift Virtualization. The
 [id="ossm-gateways_{context}"]
 == Gateways
 
-{SMProductName} installs ingress and egress gateways by default. You can disable this in the SMCP using `spec.gateways.ingress.enabled=false` or `spec.gateways.egress.enabled=false`.
+{SMProductName} installs ingress and egress gateways by default. You can disable gateway installation in the `ServiceMeshControlPlane` (SMCP) resource using the following settings:
```
Comment (tmalove) [inline modules/ossm-vs-istio.adoc:129]:
```suggestion
{SMProductName} installs ingress and egress gateways by default. You can disable gateway installation in the `ServiceMeshControlPlane` (SMCP) resource by using the following settings:
```

Comment (tmalove) [reply modules/ossm-vs-istio.adoc:129]: See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage

File: modules/ossm-vs-istio.adoc [hunk 3]
```diff
@@ -126,7 +126,16 @@ You can deploy virtual machines to OpenShift using OpenShift Virtualization. The
...
+
+* `spec.gateways.enabled=false` to disable both ingress and egress gateways.
+* `spec.gateways.ingress.enabled=false` to disable ingress gateways.
+* `spec.gateways.egress.enabled=false`  to disable egress gateways.
+
+[NOTE]
+====
```
Comment (tmalove) [inline modules/ossm-vs-istio.adoc:136]: A suggestion is to change 'are annotated' and 'are generated to the active voice, if possible to retain the meaning.

Comment (tmalove) [reply modules/ossm-vs-istio.adoc:136]: See the ISG: https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#voice

File: modules/ossm-vs-istio.adoc [hunk 4]
```diff
@@ -126,7 +126,16 @@ You can deploy virtual machines to OpenShift using OpenShift Virtualization. The
...
+The default gateways are annotated to indicate that they are generated by and managed by the {SMProductName} Operator.
```
Comment (tmalove) [inline modules/ossm-vs-istio.adoc:137]: A suggestion is to change 'are annotated' and 'are generated to the active voice, if possible to retain the meaning.

See the ISG: https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#voice

File: service_mesh/v2x/upgrading-ossm.adoc [hunk 5]
```diff
@@ -28,7 +28,9 @@ endif::[]
 [id="upgrading-control-plane"]
 == Upgrading the Service Mesh control plane
 
-You must manually update the control plane for minor and major releases.
+You must manually update the control plane for minor and major releases. While the community Istio project recommends canary upgrades, {SMProductName} only supports in-place upgrades. {SMProductName} requires that you upgrade from each minor release to the next minor release in sequence. For example, you must upgrade from version 2.0 to version 2.1, and then upgrade to version 2.2. You cannot update from {SMProductName} 2.0 to 2.2 directly.
```
Comment (tmalove) [inline service_mesh/v2x/upgrading-ossm.adoc:31]:
```suggestion
You must manually update the control plane for minor and major releases. The community Istio project recommends canary upgrades, {SMProductName} only supports in-place upgrades. {SMProductName} requires that you upgrade from each minor release to the next minor release in sequence. For example, you must upgrade from version 2.0 to version 2.1, and upgrade to version 2.2. You cannot update from {SMProductName} 2.0 to 2.2 directly.
```

**Pull request merged**